BrentOzarULTD/SQL-Server-First-Responder-Kit
# Issue 221

Fixes #221  .

Changes proposed in this pull request:
 - Added Output parameters to sp_BlitzIndex to send Mode 2 results to permanent repository

How to test this code:
 - EXEC sp_BlitzIndex @Mode=2, @OutputDatabaseName='dbName', @OutputSchemaName='schemaName', @OutputTableName='tableName', @OutputServerName='linkedServerName'
- OutputServerName is optional, the rest are all or nothing
- Works with global temp tables (##) but not local temp tables (#). No temp tables when using linked server.
- Only works on Mode 2 for now
- Should error out if DB, schema, table, or linked server are invalid or are not all provided

Has been tested on (remove any that don't apply):
 - Case-sensitive SQL Server instance
 - SQL Server 2008
 - SQL Server 2014
 - SQL Server 2016
 - SQL Server 2016 reporting to linked SQL 2008
 - SQL Server 2008 reporting to linked SQL 2016

Notes:
I hate building dynamic SQL using a bunch of string concatenation. My opinion - it makes for complicated copy/paste and sections of this will require copy/paste each time it is updated with new checks and columns. Right now it's set to do a REPLACE on variable names. I think it is easier to read and maintain. Definitely worse performance, but this isn't a high volume procedure so I don't think that matters.
Obviously that makes this pattern different than most of the other code in here though. So if it's going to bother people let me know and I'll make it fit the existing pattern.
